partiql-plan/src/main/resources/partiql_plan_0_1.ion

```diff
@@ -115,10 +118,34 @@ rex::{
       ],
     },
 
-    call::{
-       fn:    fn,
-       args:  list::[rex],
-    },
+    call::[
+      static::{
+        fn: fn,
+        args: list::[rex]
+      },
+
+      // Represents a dynamic function call. If all candidates are exhausted, dynamic calls will return MISSING.
+      //
+      // args: represent the original typed arguments. These will eventually be wrapped by coercions from [candidates].
+      // candidates: represent the potentially applicable resolved functions with coercions. Each of these candidates
+      //  should be overloaded functions of the same name and number of arguments.
+      dynamic::{
+        args: list::[rex],
+        candidates: list::[candidate],
+        _: [
+
+          // Represents a viable candidate.
+          //
+          // fn: The resolved function
+          // coercions: Represents the coercion that should take place -- if not null. The first item is to be wrapped
+          //  over the first argument of dynamic.args and so on.
+          candidate::{
+            fn: '.fn.resolved',
+            coercions: list::[optional::fn]
```

Comment:
Why not `optional::'.fn.resolved'`?

Reply:
Each candidate function is statically known, so this can't be optional. This is the equivalent of "specific".

Reply:
I meant for the coercions

Reply:
Ah, I see what you're getting at. Totally agree -- let me address, and I'll resubmit. Thanks!